test/provider.test.js

```diff
@@ -0,0 +1,100 @@
+import React, { useContext } from 'react'
+import { render } from '@testing-library/react'
+import { suppressConsoleErrors } from './support/helpers';
+
+import { RecurlyProvider } from '../lib';
+import { RecurlyContext } from '../lib/provider';
+
+describe('<RecurlyProvider />', function () {
+  describe('without having included recurly.js', function () {
+    const storedRecurly = window.recurly;
+
+    suppressConsoleErrors();
+
+    beforeEach(function () {
+      delete window.recurly;
+    });
+
+    afterEach(function () {
+      window.recurly = storedRecurly;
+    });
+
+    it('throws an error', function () {
+      expect(() => {
+        render(<RecurlyProvider publicKey="test-public-key" />);
+      }).toThrow(/Please load Recurly\.js \(https:\/\/js\.recurly\.com\/v4\/recurly\.js\) on this page/);
+    });
+  });
+
+  describe('without a publicKey', function () {
+    suppressConsoleErrors();
+
+    it('throws an error', function () {
+      expect(() => {
+        render(<RecurlyProvider />);
+      }).toThrow(/Please pass your 'publicKey' value to this RecurlyProvider/);
+    });
+  });
+
+  describe('with a publicKey', function () {
+    it('does not throw an error', function () {
+      expect(() => {
+        render(<RecurlyProvider publicKey="test-public-key" />);
+      }).not.toThrow();
+    });
+
+    it('provides a Recurly instance', function () {
+      render(
+        <RecurlyProvider publicKey="test-public-key">
+          <StubComponent />
+        </RecurlyProvider>
+      );
+
+      function StubComponent () {
+        const recurlyContext = useContext(RecurlyContext);
+        expect(recurlyContext.recurly).toBeInstanceOf(window.recurly.Recurly);
+        return '';
```

Comment:
using the `RecurlyContext` in this way causes this test to leak an abstraction from the component under test, but it is the most effective way to extract the `Recurly` instance and introspect the effect of using `<RecurlyProvider />`

Reply:
I'm not sure what you mean by "leak an abstraction", but this looks fine